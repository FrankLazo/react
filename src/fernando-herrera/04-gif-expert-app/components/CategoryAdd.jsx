import { useState } from 'react';
import PropTypes from 'prop-types';

const CategoryAdd = ({ setCategories }) =>
{
	const [inputValue, setInputValue] = useState('');

	const handleInputChange = ({ target }) =>
	{
		setInputValue(target.value);
	};

	const handleSubmit = (e) =>
	{
		e.preventDefault();

		if (inputValue.length > 0)
		{
			setCategories(cats => [inputValue, ...cats]);
			setInputValue('');
		}
	};

	return (
		<form onSubmit={ handleSubmit }>
			<div className="field">
				<div className="control">
					<input
						type="text"
						value={ inputValue }
						className="input"
						onChange={ handleInputChange }
						/>
				</div>
			</div>
		</form>
	);
};

CategoryAdd.propTypes = {
	setCategories: PropTypes.func.isRequired,
};

export default CategoryAdd;
