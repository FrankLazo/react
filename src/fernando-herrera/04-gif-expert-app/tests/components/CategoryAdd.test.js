// import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CategoryAdd from '../../components/CategoryAdd';

describe('Tests on CategoryAdd', () => {
	const setCategories = jest.fn();
	let wrapper = shallow(<CategoryAdd setCategories={ setCategories } />);

	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = shallow(<CategoryAdd setCategories={ setCategories } />);
	});

	test('should show the right component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should change text input', () => {
		const input = wrapper.find('input');
		const value = 'Hello world!';

		input.simulate('change', { target: { value: value } });
	});

	test('should not post info by submit', () => {
		wrapper.find('form').simulate('submit', { preventDefault(){} });
		expect(setCategories).not.toHaveBeenCalled();
	});

	test('should call setCategories and clean text input', () => {
		const value = 'Hello world!';

		wrapper.find('input').simulate('change', { target: { value } });
		wrapper.find('form').simulate('submit', { preventDefault(){} });

		expect(setCategories).toHaveBeenCalled();
		expect(setCategories).toHaveBeenCalledTimes(1);
		expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );

		expect(wrapper.find('input').prop('value')).toBe('');
	});
});
