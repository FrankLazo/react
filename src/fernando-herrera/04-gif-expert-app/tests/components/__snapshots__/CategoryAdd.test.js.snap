// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on CategoryAdd should show the right component 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <div
      className="control"
    >
      <input
        className="input"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
</form>
`;
